client/src/components/MultiJurisdictionAudit/AASetup/Review/index.tsx

```diff
@@ -197,110 +207,118 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
+                                  </Radio>
+                                ) : (
+                                  <Radio value={option.key} key={option.key}>
+                                    {option.key === 'asn'
+                                      ? 'BRAVO Average Sample Number: '
+                                      : ''}
+                                    {`${option.size} samples`}
+                                    {option.prob
+                                      ? ` (${percentFormatter.format(
+                                          option.prob
+                                        )} chance of reaching risk limit and completing the audit in one round)`
+                                      : ''}
+                                  </Radio>
+                                )
+                              }
                             )}
-                          />
-                        )}
-                      </FormSectionDescription>
-                    </ElevatedCard>
-                  )
-                })}
-              </FormSection>
-            )}
-            <FormButtonBar>
-              <FormButton onClick={prevStage.activate}>Back</FormButton>
-              <FormButton
-                intent="primary"
-                disabled={
-                  locked ||
-                  !isSetupComplete(jurisdictions, contests, auditSettings)
-                }
-                onClick={handleSubmit}
-              >
-                Launch Audit
-              </FormButton>
-            </FormButtonBar>
-          </Form>
-        )}
-      </Formik>
+                          </RadioGroup>
+                          {currentOption && currentOption.key === 'custom' && (
+                            <Field
+                              component={FormField}
+                              name={`sampleSizes[${contest.id}][size]`}
+                              type="text"
+                              validate={testNumber(
+                                Number(contest.totalBallotsCast),
+                                `Must be less than or equal to: ${contest.totalBallotsCast} (the total number of ballots in this targeted contest)`
+                              )}
+                            />
+                          )}
+                        </FormSectionDescription>
+                      </ElevatedCard>
+                    )
+                  })}
+                </FormSection>
+              )}
+              <FormButtonBar>
+                <FormButton onClick={prevStage.activate}>Back</FormButton>
+                <FormButton
+                  intent="primary"
+                  disabled={
+                    locked ||
+                    !isSetupComplete(jurisdictions, contests, auditSettings)
+                  }
+                  onClick={handleSubmit}
+                >
+                  Launch Audit
+                </FormButton>
+              </FormButtonBar>
```

Comment:
does this hide the launch audit button if the tallies files haven't been uploaded? maybe it would be better to just disable it

Reply:
@jonahkagan It hides it currently. The reason (from what I remember) is because the information for rendering the form is dependent on the return from the sample size option calculation, which can't be called if the tallies files haven't been uploaded. The launch audit button is also the submit button for the form, so it's dependent on the form being there.

Reply:
could you conditionally render the content of the form that depends on the sample size options and leave the button?

Reply:
@jonahkagan Success! I was able to make that work, and along the way I found some bugs that I was able to fix as well, haha.